docs/proposals/cascading-deletion.md

```diff
@@ -0,0 +1,157 @@
+# Overview
+Currently cascading deletion is implemented at client-side. For example, when deleting a replica set, kubectl uses a reaper to delete the created pods and then delete the replica set. We plan to move the cascading deletion to the server to simplify the client-side logic. In this proposal, we present two alternative designs. [Design #1](#design-1-tombstone--gc) uses the tombstone mechanism, and [Desig #2](#design-2-finalizer-framework--gc) uses the finalizer mechanism.
+
+# Goals
+* Supporting cascading deletion at the server-side. 
+* Centralizing the cascading deletion logic, rather than spreading in controllers.
+* Allowing optionally orphan the dependent objects.
+
+# Non goals
+* Releasing the name of an object immediately, so it can be reused ASAP.
+
+# Design 1. Tombstone + GC
+## API Changes
+```
+type DeleteOptions struct {
+	…
+	Orphaning bool
```

Comment:
Do we have a clear parent child relationship defined somewhere?

Do we need something like controllerRef to materalize the same basic thing we have today for `ObjectMeta.Namespace` so the relationship is explicit?